### PR TITLE
Added the option for using custom setting for docker. 

### PR DESCRIPTION
**Pull request by KaiserBh**
.

### CODE REVIEW

**Comment by KaiserBh** [comment]
#73

File: Docker/PlexAniSync/Dockerfile [hunk 1]
```diff
@@ -10,14 +10,30 @@ ENV PLEX_SECTION=Anime \
...
+WORKDIR /plexanisync
+
+# Adding requirements file to current directory
+# just this file first to cache the pip install step when code changes
+# Use copy since we are not getting anything from remote or URL
+COPY requirements.txt .
+COPY Docker/PlexAniSync/run/. .
+
+# Set work directory early so remaining paths can be relative
+WORKDIR /plexanisync
```
Comment (reconman) [inline Docker/PlexAniSync/Dockerfile:30]: Duplicate of lines 20 and 21

File: Docker/PlexAniSync/README.md [hunk 2]
```diff
@@ -19,19 +20,34 @@ docker run -d \
...
+
+
+  ghcr.io/rickdb/plexanisync:latest
+```
+
 ### Environment Variables
 
-| ID                          | Default                  | Required | Note                                                                                                                                                                         |
-| --------------------------- | ------------------------ | :------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| PLEX_SECTION                | Anime                    | &#10003; | The library where your anime resides                                                                                                                                         |
```
Comment (reconman) [inline Docker/PlexAniSync/README.md:26]: I would mark the required fields with an asterisk and then write below the table that they're not required if you provide a custom settings file.

Additionally, `runsync.sh` should only call `settingsupdater.py` if the SETTINGS_FILE environment variable was not set.

File: PlexAniSync.py [hunk 3]
```diff
@@ -51,11 +51,19 @@ def read_settings(settings_file) -> configparser.ConfigParser:
     return settings
 
 
-if len(sys.argv) > 1:
-    SETTINGS_FILE = sys.argv[1]
-    logger.warning(f"Found settings file parameter and using: {SETTINGS_FILE}")
+# check if setting parameters are set.
+# check if the arg given is empty
+check_settings_arg = os.environ.get("SETTINGS_FILE")
```
Comment (reconman) [inline PlexAniSync.py:56]:
```suggestion
SETTINGS_FILE = os.getenv("SETTINGS_FILE", "settings.ini")
```

Then almost all the following lines can be removed as only the old `sys.argv[1]` code is needed anymore.

File: Docker/PlexAniSync/Dockerfile [hunk 4]
```diff
@@ -10,14 +10,30 @@ ENV PLEX_SECTION=Anime \
     ANI_USERNAME='' \
     ANI_TOKEN='' \
     INTERVAL=3600 \
-    PATH="${PATH}:~/.local/bin"
+    SETTINGS_FILE=''
```
Comment (reconman) [inline Docker/PlexAniSync/Dockerfile:13]: All the changes to the Docker files and python script should also be applied to the Tautulli ones.

**Review by reconman — CHANGES_REQUESTED** [review]

File: PlexAniSync.py (hunk 3 above)
Comment (KaiserBh) [reply PlexAniSync.py:56]: > Then almost all the following lines can be removed as only the old `sys.argv[1]` code is needed anymore.
This doesn't seem to work, I tried it but it fails to find `settings.ini` for some reason even tho it calls `settingsupdater.py`

**Pull request closed without merging**